CMakeLists.txt

```diff
@@ -1,25 +1,23 @@
-cmake_minimum_required(VERSION 3.2)
-
-option(JSON_VALIDATOR_BUILD_TESTS    "Build tests"    ON)
-option(JSON_VALIDATOR_BUILD_EXAMPLES "Build examples" ON)
-option(JSON_VALIDATOR_HUNTER "Enable Hunter package manager support" OFF)
-
-if(JSON_VALIDATOR_HUNTER)
-    include("cmake/HunterGate.cmake")
-    HunterGate(
-        URL "https://github.com/cpp-pm/hunter/archive/v0.23.262.tar.gz"
-        SHA1 "eb51e633e08cdbe2153caf255e9c23968fecb29d"
-    )
-endif()
+cmake_minimum_required(VERSION 3.16)
+
+option(JSON_VALIDATOR_BUILD_TESTS    "Build tests"    OFF)
+option(JSON_VALIDATOR_BUILD_EXAMPLES "Build examples" OFF)
 
 # the project
 project(nlohmann_json_schema_validator
         LANGUAGES CXX)
 
-set(PROJECT_VERSION 2.1.1)
-
-if(JSON_VALIDATOR_HUNTER)
-    hunter_add_package(nlohmann_json)
+set(PROJECT_VERSION 2.1.2)
+include(FetchContent)
+
+# find nlohmann_json if not found simply fetch it
+find_package(nlohmann_json QUIET)
+if(NOT nlohmann_json_FOUND)
+    message(STATUS "nlohmann_json NOT FOUND, fetching version v3.10.4")
+    FetchContent_Declare(json
+    GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent
```

Comment:
Why is it ArthurSonzogni for nlohmann-json but below pboettch for json-schema-validator?

Reply:
Just a matter of size. We can do the same with `nlohmann_json` but the fetch is "big", this one automates the extraction of the header only part, and does not take care of the rest ... but I do not care if we want to stick we the official one.

Reply:
Ah, I didn't realize that FetchContent fetch the git, I thought by specifying the version it will get the release tar.gz

Reply:
Can't we fetch nlohmann with 
```CMake
FetchContent_Declare(
  myCompanyIcons
  URL      https://intranet.mycompany.com/assets/iconset_1.12.tar.gz
  URL_HASH MD5=5588a7b18261c20068beabfb4f530b87
)
```

Reply:
run some tests, building from scratch:

```
# nlohmann_json
cmake -B build  19.74s user 3.92s system 49% cpu 47.607 total
# nlohmann_json_cmake_fetchcontent
cmake -B build  0.60s user 0.35s system 22% cpu 4.159 total
```

Reply:
I don't understand the differences.

Reply:
First one fetch actual `nlohmann_json` repo --> take 47 sec to build
Second one fetch `nlohmann_json_cmake_fetchcontent` repo --> take 4 sec to build

So fetching actual `nlohmann_json` multiply the **first** build duration by 10 for people that just want to include `json-schema-validator` in the project and do not care about how `nlohmann_json` is provisioned.

That all, we can use one of the other, I do not care only difference is the first build duration.